cloud_functions/climateiq_data_export_cf/main.py

```diff
@@ -302,20 +347,152 @@ def _calculate_h3_indexes(
         ["h3_index", "h3_centroid_lat", "h3_centroid_lon", "h3_boundary"]
     ] = spatialized_predictions.apply(_add_h3_index_details, axis=1)
 
-    # Filter out any cells where the projected H3 centroid falls outside of the
+    # Filter out any rows where the projected H3 centroid falls outside of the
     # chunk boundary.
     chunk_boundary = _get_chunk_boundary(study_area_metadata, chunk_metadata)
     spatialized_predictions = spatialized_predictions[
         spatialized_predictions.apply(
-            lambda cell: chunk_boundary.contains(
-                geometry.Point(cell["h3_centroid_lon"], cell["h3_centroid_lat"])
+            lambda row: chunk_boundary.contains(
+                geometry.Point(row["h3_centroid_lon"], row["h3_centroid_lat"])
             ),
             axis=1,
         )
     ]
 
-    # TODO: Add logic for fetching predictions from neighboring chunks for boundary
-    # cells.
+    # Extract rows where the projected H3 cell is not fully contained within the chunk
+    # so we can aggregate prediction values across chunk boundaries.
+    boundary_h3_cells = spatialized_predictions[
+        spatialized_predictions.apply(
+            lambda row: not row["h3_boundary"].within(chunk_boundary),
+            axis=1,
+        )
+    ]["h3_boundary"].unique()
+
+    return _aggregate_h3_predictions(
+        study_area_metadata,
+        chunk_metadata,
+        boundary_h3_cells,
+        spatialized_predictions,
+        bucket_name,
+        object_name,
+    )
+
+
+def _aggregate_h3_predictions(
+    study_area_metadata: dict,
+    chunk_metadata: dict,
+    boundary_h3_cells: np.ndarray,
+    spatialized_predictions: pd.DataFrame,
+    bucket_name: str,
+    object_name: str,
+) -> pd.Series:
+    """Aggregates predictions for duplicate H3 projections across chunk boundaries.
+
+    Args:
+        study_area_metadata: A dictionary containing metadata for the study
+        area.
+        chunk_metadata: A dictionary containing metadata for the chunk.
+        boundary_h3_cells: An array containing the H3 boundaries of cells that
+        intersect with the chunk boundary.
+        spatialized_predictions: A DataFrame containing H3 indexes and predictions
+        for a single chunk. This input DataFrame may contain duplicate H3 indexes.
+        bucket_name: The name of the GCS bucket containing model predictions.
+        object_name: The name of the chunk object this cloud function is currently
+        processing. Used to construct the GCS object name of the neighbor chunk.
+
+    Returns:
+        A Series containing H3 indexes in a single chunk along with associated
+        predictions, representing a subset of the full study area results.
+    """
+    x = chunk_metadata["x_index"]
+    y = chunk_metadata["y_index"]
+
+    # Process direct 8 neighbors of current chunk. Since H3 cells at resolution 13
+    # (43.870 m^2)have a smaller area than chunks (), it is guaranteed that any
```

Comment:
@waltaskew do you know what the area of the chunks are?

Reply:
two thousand meters squared

Reply:
thanks!